pkg/execution/executor/lifecycle.go

```diff
@@ -0,0 +1,100 @@
+package executor
+
+import (
+	"context"
+	"log/slog"
+	"strings"
+	"time"
+
+	"github.com/inngest/inngest/pkg/consts"
+	"github.com/inngest/inngest/pkg/enums"
+	"github.com/inngest/inngest/pkg/execution"
+	sv2 "github.com/inngest/inngest/pkg/execution/state/v2"
+	"github.com/inngest/inngest/pkg/telemetry"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func NewLifecycleListener(l *slog.Logger, run sv2.RunService) execution.LifecycleListener {
```

Comment:
`executor.NewLifecycelListener()` is quite generic - would it be reasonable for this to be more attributed to tracing? It could still be in the `executor` package if needed.